src/compose/network.ts

```diff
@@ -90,19 +80,26 @@ export class Network {
 
 		Network.validateComposeConfig(network);
 
-		const ipam: Partial<ComposeNetworkConfig['ipam']> = network.ipam || {};
-		if (ipam.driver == null) {
-			ipam.driver = 'default';
-		}
-		if (ipam.config == null) {
-			ipam.config = [];
-		}
-		if (ipam.options == null) {
-			ipam.options = {};
-		}
+		const ipam = network.ipam ?? {};
+		const driver = ipam.driver ?? 'default';
+		const config = ipam.config ?? [];
+		const options = ipam.options ?? {};
+
 		net.config = {
 			driver: network.driver || 'bridge',
-			ipam: ipam as ComposeNetworkConfig['ipam'],
+			ipam: {
+				driver,
+				config: config.map((conf) => ({
+					...(conf.subnet && { subnet: conf.subnet }),
+					...(conf.gateway && { gateway: conf.gateway }),
+					...(conf.ip_range && { ipRange: conf.ip_range }),
+					// TODO: compose defines aux_addresses as a dict but dockerode and the
+					// engine accepts a single AuxAddress. What happens when multiple addresses
+					// are given?
+					...(conf.aux_addresses && { auxAddress: conf.aux_addresses }),
```

Comment:
The engine should [accept](https://docs.docker.com/engine/reference/commandline/network_create/#specify-advanced-options) multiple aux addresses, so I assume Dockerode does too - do the Dockerode typings not reflect this?

Reply:
The engine should, but it looks like the dockerode typings go by what is defined on the API https://docs.docker.com/engine/api/v1.37/#operation/NetworkCreate